src/cpp/shared_core/include/shared_core/json/Json.hpp

```diff
@@ -1790,7 +1790,7 @@ template <typename T>
 inline Value toJsonValue(const std::vector<T>& in_vector)
 {
    Array results;
-   for (const T& val : in_vector)
+   for (auto&& val : in_vector)
```

Comment:
This compiles? The parameter is a `const &`, so I think we should use `const &` here to ensure that we won't accidentally gut something we shouldn't.

Reply:
It does -- this actually does deduce as `const T&` for a regular `const std::vector<T>&`. This is because it basically uses the result of `decltype(*val.begin())`, and for const vectors this will return a const interator which will yield a `const T&` when dereferenced. See e.g.

https://godbolt.org/z/Ej973W

In other words, `auto&&` preserves the const-ness here because the container itself handles const-ness in its iterators.

Reply:
In that case, why change this away from `const T& val` to `auto&&` when we use `const auto&` in other places? I also think both `const auto&` and `const T&` are more obvious in their meaning to the reader.

Reply:
Oh -  I see in the approach section it's because of the potential weirdness with `std::vector<bool>`